Docs: add new LLM providers (agentgateway 1.3) (#636)
agentgateway/website: content/docs/kubernetes/main/llm/providers/custom.md

---
title: Custom providers
weight: 22
description: Configure unsupported, self-hosted, and non-standard LLM providers with explicit API formats, paths, and backend targets.
test: skip
---

Use custom providers for unsupported, self-hosted, or non-standard LLM
targets when you want to declare the provider target and supported API formats
explicitly.

> If your upstream already matches a first-class provider page, or it is a

Reviewer: ```suggestion
If your upstream already matches a first-class provider page, or the provider is 
```

> generic OpenAI-compatible provider without special path or format handling,

Reviewer: ```suggestion
generic OpenAI-compatible without special path or format handling,
```

> prefer those simpler options first.

Reviewer: ```suggestion
use the standard provider guides instead of this custom provider guide.
```
What does simpler options mean here? Like the docs for that specific provider?

Author: it just means do the standard way vs this custom guide, rephrasing

  {{< reuse "agw-docs/snippets/backend.md" >}} that routes to a Kubernetes
  Service or InferencePool.

For first-class providers such as OpenAI, Anthropic, Gemini, Vertex AI, Azure,
Bedrock, and Ollama, use the dedicated provider page unless you need explicit
format or backend target control. For providers that expose the standard OpenAI
API shape without a first-class type, such as Cohere, DeepSeek, Groq, Mistral,

> Together AI, and xAI, use [OpenAI-compatible providers]({{< link-hextra path="/llm/providers/openai-compatible/" >}}) before reaching for a custom provider.

Reviewer: ```suggestion
Together AI, and xAI, use [OpenAI-compatible providers]({{< link-hextra path="/llm/providers/openai-compatible/" >}}) before using a custom provider.
```
Is it a custom provider or a custom provider configuration?

Author: What do you think the difference would be between those two things? Custom provider is a way for you to customize a provider via a certain configuration. But even a standard/first-class provider, you still have configuration for.

Reviewer: I wasn't sure if it was talking about different configuration docs or a different provider entirely.
